### PR TITLE
Populate object via page params

### PR DESCRIPTION
**Pull request by michaelatias**

### CODE REVIEW

File: client-side/src/app/data-configuration-block/data-configuration-block.component.ts [hunk 1]
```diff
@@ -52,6 +61,9 @@ export class DataConfigurationBlockComponent implements OnInit {
     generateDataViewFormFields(): DataView[]{
       return this.fields = this.hostObject.configuration.cardsList.map(card => card.value)
     }
-    onValueChanged($event){
+    async updateItem(resourceName: string, key: string){
+      const items = await this.udcService.getItems(resourceName)
```
Comment (roiaharon-pepperi) [inline client-side/src/app/data-configuration-block/data-configuration-block.component.ts:65]: why do you get the items each update? the items does not change that much. do it once when a resource changes

File: client-side/src/app/data-configuration-block/data-configuration-block.component.ts [hunk 2]
```diff
@@ -27,11 +29,18 @@ export class DataConfigurationBlockComponent implements OnInit {
...
-    constructor(private translate: TranslateService) {
+    constructor(private translate: TranslateService, private udcService: UDCService) {
+      this.udcService.pluginUUID = config.AddonUUID
       this.typeMap = new TypeMap()
       this.typeMap.init()
     }
     ngOnInit(): void {
+      if(this.hostObject?.parameters){
+        const resourceAndKey = this.hostObject?.parameters['resource_key'] || "";
+        const [resourceName, key] = resourceAndKey.split(" ")
```
Comment (roiaharon-pepperi) [inline client-side/src/app/data-configuration-block/data-configuration-block.component.ts:40]: space is not a good delimiter. both resource and key can contains it. maybe go with '@'

File: client-side/src/app/data-configuration-block-editor/data-configuration-block-editor.component.ts [hunk 3]
```diff
@@ -147,4 +148,19 @@ export class DataConfigurationBlockEditorComponent implements OnInit {
...
         this.updateAllConfigurationObject()
     }
+    setPageConfiguration(){
+        this.hostEvents.emit({
+            action: 'set-page-configuration',
+            
+            pageConfiguration: {
+                Parameters: [
+                    {
+                        Key: "resource_key",
```
Comment (roiaharon-pepperi) [inline client-side/src/app/data-configuration-block-editor/data-configuration-block-editor.component.ts:158]: there is no value? where do you add value to this parameter?

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/data-configuration-block-editor/data-configuration-block-editor.component.ts (hunk 3 above)
Comment (michaelatias) [reply client-side/src/app/data-configuration-block-editor/data-configuration-block-editor.component.ts:158]: the value of this parameter will given to me via the url

File: client-side/src/app/data-configuration-block/data-configuration-block.component.ts (hunk 2 above)
Comment (roiaharon-pepperi) [reply client-side/src/app/data-configuration-block/data-configuration-block.component.ts:40]: ok, no problem

File: client-side/src/app/data-configuration-block/data-configuration-block.component.ts (hunk 1 above)
Comment (roiaharon-pepperi) [reply client-side/src/app/data-configuration-block/data-configuration-block.component.ts:65]: ok. no problem

**Review by roiaharon-pepperi — APPROVED** [review]

**Pull request merged**